src/compiler/executable.cc

```diff
@@ -48,10 +48,14 @@ static const uint8 VESSEL_TOKEN[] = { VESSEL_TOKEN_VALUES };
 // just much more complicated for something that doesn't change that frequently.
 static int VESSEL_SIZES[] = { 128, 256, 512, 1024, 8192, };
 
-static int sign_if_necessary(const char* out_path) {
+static int sign_if_necessary(const char* out_path, const char* arch) {
 #ifndef TOIT_DARWIN
+  // TODO(florian): sign if arch equals "macos".
```

Comment:
macos isn't an architecture?

Typically, architecture will be something like arm64 and operating system will be macos, linux, win.